File: _etc/Java/11726 2xn 타일링.java
import java.util.Scanner;

public class Main {
    public static void main(String args[]) {
      Scanner s = new Scanner(System.in);
      int n = s.nextInt();
      long[] dp = new long[1001];
      
      dp[0] = 0;
      dp[1] = 1;
      dp[2] = 2;
      dp[3] = 3;
      
      for(int i = 4; i < n+1; i++) {
          dp[i] = (dp[i-1] + dp[i-2])%10007;
      }
      
      System.out.println(dp[n]);
      
    }
}

// http://boj.kr/63982614b20540768bd69ac512053d92

File: mvc1/_etc/Kotlin/11726 2xn 타일링.kt
import java.util.Scanner
import kotlin.math.*

fun main(args: Array<String>) {
    val sc: Scanner = Scanner(System.`in`)
    val num = sc.nextInt()
    
    var dp = LongArray(1001, {0})
    
    dp[0] = 0
    dp[1] = 1
    dp[2] = 2
    dp[3] = 3
    
    for(i in 4..num) {
        dp[i] = (dp[i-1] + dp[i-2]) % 10007
    }
    
    println("${dp[num]}")

}

// http://boj.kr/67b7c599a4af4274afd983bdd8143b5f
